client/shipping-label/state/actions.js

```diff
@@ -487,6 +492,9 @@ export const purchaseLabel = () => ( dispatch, getState, context ) => {
 			} else if ( error ) {
 				console.error( error );
 				dispatch( NoticeActions.errorNotice( error.toString() ) );
+				//re-request the rates on failure to avoid attempting repurchase of the same shipment id
+				dispatch( clearAvailableRates() );
+				getLabelRates( dispatch, getState, _.noop, { getRatesURL, nonce } );
```

Comment:
Is there any opportunity to selectively refresh rates depending on the type of error?

In the case where our server can't successfully reach the API for rates/labels - would a refresh be necessary? Couldn't we just try the same purchase request(s) again?